java/address-java/src/main/java/uk/os/elements/address/utils/PostcodeUtil.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2016 Ordnance Survey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.os.elements.address.utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Provide common utility functions for postcodes.
+ *
+ * <p>
+ * Note:
+ *
+ * <ul>
+ * <li>The Royal Mail is the data authority
+ * <ul>
+ * <li>they reserve the right to change their data at any time
+ * </ul>
+ * </li>
+ * <li>the only way to establish if a postcode is valid is to check the source postcode data
+ * <li>these utility functions are only checked against current postcodes; historical and future postcodes may fail these tests.
+ * </ul>
+ *
+ * <p>
+ * Breakdown of a postcode.
+ *
+ * <p>
+ * Royal Mail (2016, p.18) provide the following example:
+ *
+ * Using postcode "PO1 1AF" the components are:
+ * 'PO'1 1AF  - postcode area
+ *  PO'1' 1AF - postcode district
+ *  PO1 '1'AF - postcode sector
+ *  PO1 1'AF' - unit postcode
```

Comment:
Would it be less likely to be misunderstood if we just listed these as: area, district, sector, unit?